lib/floe/workflow/states/pass.rb

```diff
@@ -25,7 +25,8 @@ def initialize(workflow, name, payload)
         end
 
         def finish(context)
-          context.output = process_output(context, result)
+          input = result.nil? ? process_input(context) : result
+          context.output = process_output(context, input)
```

Comment:
This makes all my pass tests pass.

But I think the fix belongs in `process_output`

```diff
        def process_output(context, results)
-          return context.input.dup if results.nil?
+          return process_input(context) if results.nil?
```

Not sure if we need a dup there or not.

Reply:
I think doing `context.output = process_input(context) if results.nil?` would bypass the `OutputPath` part no?

Reply:
I added a spec showing why this doesn't work with OutputPath, https://github.com/ManageIQ/floe/pull/225/files#diff-c4b20f6026d80247d4c5645a2811f66f0bd918ebb4763da5fb0d19c8a7d092faR115-R123

Reply:
this did bypass output processing. So the solution would be:

```ruby
results ||= process_input(context
```

BUT, per an out of band conversation with @agrare:

> Let's not put input processing into output processing.

So the preferred implementation puts the fixing of `result.nil?` on the caller:

```ruby
process_output(context, results || process_input(context)
```

Reply:
Is that just a one-line version of what I have ?

Reply:
Yes. that was from you. and I was trying to bring that conversation back here